bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -150,18 +155,16 @@ void KeyExchangeManager::loadClientPublicKeys() {
 }
 
 void KeyExchangeManager::exchangeTlsKeys(const std::string& type, const SeqNum& bft_sn) {
-  auto keys = concord::util::crypto::Crypto::instance().generateECDSAKeyPair(
-      concord::util::crypto::KeyFormat::PemFormat, concord::util::crypto::CurveType::secp384r1);
+  auto keys = Crypto::instance().generateECDSAKeyPair(concord::util::crypto::KeyFormat::PemFormat,
+                                                      concord::util::crypto::CurveType::secp384r1);
   bool use_unified_certs = bftEngine::ReplicaConfig::instance().useUnifiedCertificates;
   const std::string base_path =
       bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" + std::to_string(repID_);
   std::string root_path = (use_unified_certs) ? base_path : base_path + "/" + type;
-
   std::string cert_path = (use_unified_certs) ? root_path + "/node.cert" : root_path + "/" + type + ".cert";
-  std::string prev_key_pem = concord::util::crypto::Crypto::instance()
-                                 .RsaHexToPem(std::make_pair(SigManager::instance()->getSelfPrivKey(), ""))
-                                 .first;
-  auto cert = concord::util::crypto::CertificateUtils::generateSelfSignedCert(cert_path, keys.second, prev_key_pem);
+  std::string prev_key_pem =
+      Crypto::instance().RsaHexToPem(std::make_pair(SigManager::instance()->getSelfPrivKey(), "")).first;
```

Comment:
SigManager::instance()->getSelfPrivKey() would return an EdDSA key right? In that case, are we planning to use eddsa key to generate self signed cert?

Reply:
Yes, we will use EdDSA key instead of RSA key to generate self signed certificate.
However, the certificate will continue to have ECDSA key.